### PR TITLE
Instructions to resolve TLS v1.0 error on macOS

### PR DESCRIPTION
**Pull request by dmlemos**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [X] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Before I was using system python and noticed `youtube-dl` was complaining about the TLS protocol version on certain websites:

```sh
$ youtube-dl --verbose https://cauchemar.bandcamp.com/album/chapelle-ardente
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'--verbose', u'https://cauchemar.bandcamp.com/album/chapelle-ardente']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2017.09.24
[debug] Python version 2.7.10 - Darwin-16.7.0-x86_64-i386-64bit
[debug] exe versions: ffmpeg 3.3.4, ffprobe 3.3.4
[debug] Proxy map: {}
[Bandcamp:album] chapelle-ardente: Downloading webpage
[download] Downloading playlist: Chapelle ardente
[Bandcamp:album] playlist Chapelle ardente: Collected 9 video ids (downloading 9 of them)
[download] Downloading video 1 of 9
[Bandcamp] n-cromancie: Downloading webpage
[debug] Default format spec: bestvideo+bestaudio/best
[debug] Invoking downloader on u'https://t4.bcbits.com/stream/f2cd5386df394ca9ea7356be56707148/mp3-128/4099989480?p=0&ts=1506596271&t=ac53a5efdcce9856b2a44e28bc8c6074948da14d&token=1506596271_d948460a3fb21d71fd4965a2f0f07affe97fa509'
ERROR: unable to download video data: <urlopen error [SSL: TLSV1_ALERT_PROTOCOL_VERSION] tlsv1 alert protocol version (_ssl.c:590)>
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 1875, in process_info
    success = dl(filename, info_dict)
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 1814, in dl
    return fd.download(name, info)
  File "/usr/local/bin/youtube-dl/youtube_dl/downloader/common.py", line 361, in download
    return self.real_download(filename, info_dict)
  File "/usr/local/bin/youtube-dl/youtube_dl/downloader/http.py", line 286, in real_download
    establish_connection()
  File "/usr/local/bin/youtube-dl/youtube_dl/downloader/http.py", line 74, in establish_connection
    ctx.data = self.ydl.urlopen(request)
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 2178, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 431, in open
    response = self._open(req, data)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 449, in _open
    '_open', req)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/usr/local/bin/youtube-dl/youtube_dl/utils.py", line 1086, in https_open
    req, **kwargs)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 1197, in do_open
    raise URLError(err)
URLError: <urlopen error [SSL: TLSV1_ALERT_PROTOCOL_VERSION] tlsv1 alert protocol version (_ssl.c:590)>
```

Upon investigating, python was using an older openssl version:
```sh
$ python -c 'import ssl; print(ssl.OPENSSL_VERSION)'
OpenSSL 0.9.8zh 14 Jan 2016
```

This pull request provides instructions on how to fix this error on macOS.

Should resolve these issues #13936 #12976 #3731

### CODE REVIEW

**Comment by Wowfunhappy** [comment]
I am very, very wary about modifying Apple's system python in any way, as it's used by native macOS apps that expect the default to be installed. This is why Homebrew doesn't touch built in utilities.

I would love another workaround. For now, I guess I'm stuck manually updating :(

**Comment by g8keepa** [comment]
This resolution is not working.  Please advise?

**Comment by Wowfunhappy** [comment]
What I've been doing, everywhere I used to do youtube-dl --update, I instead do:

curl -L http://yt-dl.org/downloads/latest/youtube-dl -o /usr/local/bin/youtube-dl
chmod a+rx /usr/local/bin/youtube-dl

This a huge disadvantage in that a new copy will be downloaded regardless of whether or not there's an update available, but what can you do?

**Comment by g8keepa** [comment]
@Wowfunhappy that's how I always update, doesn't fix this issue at all.

**Comment by dmlemos** [comment]
@Wowfunhappy `Brew` is not modifying the system. I suggest you read what it does [here](https://brew.sh/).
The python shipped with the OS is **old**. The `brew` package manager, allows you to have `python2` and `python3` installed, while coexisting with the system python (without touching it).

I guarantee you, that once you have `brew` managing your python(s), you won't have a need to use the system one. In fact, if you are not careful with it, you are in for trouble. Still, if you intend to only have the old system python for some reason, at least use `virtualenv` and then `pip` to install it.
I prefer to have the system package manager installing/managing my packages whenever I can, falling back to `virtualenv` only out of necessity.

Now, in order to have `youtube-dl` accessible in the shell, you need to modify the `PATH` to have the brew managed python before the system one (as indicated in the PR). This is only modified in your profile. If Apple or any other dependency needs the system python, it's still there and will be the first available (they won't even know the other python exists).

Upgrade using brew:
`brew upgrade youtube-dl`

**Comment by Wowfunhappy** [comment]
`youtube-dl --update` is now working successfully on my machine. I did not follow the workaround instructions or (consciously) make any changes to how ssl is handled.

Was this issue fixed by youtube-dl and/or Apple? I'm running macOS 10.13.5.

File: README.md [hunk 1]
```diff
@@ -32,10 +32,10 @@ Windows users can [download an .exe file](https://yt-dl.org/latest/youtube-dl.ex
 You can also use pip:
 
     sudo -H pip install --upgrade youtube-dl
-    
+
 This command will update youtube-dl if you have already installed it. See the [pypi page](https://pypi.python.org/pypi/youtube_dl) for more information.
 
-macOS users can install youtube-dl with [Homebrew](https://brew.sh/):
```
Comment (timaschew) [inline README.md]: Why did you remove the link?

**Comment by timaschew** [comment]
I can confirm this error happens with python Python 2.7.10 and built-in OpenSSL 0.9.8zh.  
Upgrading to to Python 2.7.16 (and built-in OpenSSL 1.0.2s) works fine.
